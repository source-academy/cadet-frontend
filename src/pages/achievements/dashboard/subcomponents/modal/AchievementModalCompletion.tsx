import React from 'react';

type AchievementModalCompletionProps = {
  awardedExp: number;
  completionText: string;
};

function AchievementModalCompletion(props: AchievementModalCompletionProps) {
  const { awardedExp, completionText } = props;

  return (
    <div className="completion">
      <h1>{`AWARDED ${awardedExp}XP`}</h1>
      <p>{completionText}</p>
    </div>
  );
}

export default AchievementModalCompletion;
