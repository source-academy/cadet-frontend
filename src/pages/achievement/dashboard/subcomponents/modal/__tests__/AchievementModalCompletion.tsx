import { mount } from 'enzyme';
import React from 'react';

import AchievementModalCompletion from '../AchievementModalCompletion';

const mockProps = {
  awardedExp: 0,
  completionText: 'string'
};

test('AchievementModalCompletion component renders correctly', () => {
  const sampleComponent = <AchievementModalCompletion {...mockProps} />;
  const tree = mount(sampleComponent);
  expect(tree.debug()).toMatchSnapshot();
});
