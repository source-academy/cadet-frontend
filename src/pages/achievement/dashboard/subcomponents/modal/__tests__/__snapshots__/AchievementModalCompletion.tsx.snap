// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementModalCompletion component renders correctly 1`] = `
"<AchievementModalCompletion awardedExp={0} completionText=\\"string\\">
  <div className=\\"completion\\">
    <h1>
      AWARDED 0XP
    </h1>
    <p>
      string
    </p>
  </div>
</AchievementModalCompletion>"
`;
